network.go

```diff
@@ -832,35 +832,84 @@ func (n *network) updateSvcRecord(ep *endpoint, localEps []*endpoint, isAdd bool
 	}
 
 	c := n.getController()
-	sr, ok := c.svcDb[n.ID()]
-	if !ok {
+	c.Lock()
```

Comment:
Actually you can even remove the initialization of the svcDb add add it in `addSvcRecords` since it is not needed here at all. One less place to acquire c.Lock()